include/bluetooth/mesh/vnd/dm_cli.h

```diff
@@ -85,7 +85,7 @@ struct bt_mesh_dm_cli_handlers {
 	 */
 	void (*const result_handler)(struct bt_mesh_dm_cli *cli,
 				     struct bt_mesh_msg_ctx *ctx,
-				     const struct bt_mesh_dm_cli_result *result);
+				     const struct bt_mesh_dm_cli_results *result);
```

Comment:
```suggestion
				     const struct bt_mesh_dm_cli_results *results);
```

and also update function doc to use plural